spring-cloud-gcp-kms/src/main/java/com/google/cloud/spring/kms/KmsTemplate.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.kms;
+
+import java.nio.charset.Charset;
+import java.util.Base64;
+
+import com.google.cloud.kms.v1.CryptoKeyName;
+import com.google.cloud.kms.v1.DecryptRequest;
+import com.google.cloud.kms.v1.DecryptResponse;
+import com.google.cloud.kms.v1.EncryptRequest;
+import com.google.cloud.kms.v1.EncryptResponse;
+import com.google.cloud.kms.v1.KeyManagementServiceClient;
+import com.google.cloud.spring.core.GcpProjectIdProvider;
+import com.google.common.hash.Hashing;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Int64Value;
+
+/**
+ * Offers convenience methods for performing common operations on KMS including
+ * encrypting and decrypting text.
+ *
+ * @author Emmanouil Gkatziouras
+ */
+public class KmsTemplate implements KmsOperations {
+
+	private final KeyManagementServiceClient client;
+
+	private final GcpProjectIdProvider projectIdProvider;
+
+	public KmsTemplate(
+			KeyManagementServiceClient keyManagementServiceClient,
+			GcpProjectIdProvider projectIdProvider) {
+		this.client = keyManagementServiceClient;
+		this.projectIdProvider = projectIdProvider;
+	}
+
+	@Override
+	public byte[] encryptText(String cryptoKey, String text) {
+		byte[] bytes = text.getBytes(Charset.defaultCharset());
```

Comment:
We should decode using `UTF-8`, since that's the convention. Or, we can just document that the platform's default charset would be used.
```suggestion
		byte[] bytes = text.getBytes(StandardCharsets.UTF_8);
```

Reply:
Switched to StandardCharsets.UTF_8